### ISSUE TITLE
No default phase/section generated when creating a template

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#191]
If you create a new template it does not generate a default phase or section, so you are unable to proceed further with the template creation process.

![screen shot 2017-03-21 at 2 35 51 pm](https://cloud.githubusercontent.com/assets/1204467/24171958/bd84f312-0e43-11e7-824e-726df14bc994.png)

No errors appear in the log

**Comment by briri** [comment]
Updated the template_controller to create a stub 'Phase 1' when you create a template. 

This allowed the buttons to appear now on the details tab.
![screen shot 2017-03-21 at 3 11 51 pm](https://cloud.githubusercontent.com/assets/1204467/24173310/cb4c6a98-0e48-11e7-839a-94b545fa8f4d.png)

However, when you click on the phase tab, there are no options to add sections
![screen shot 2017-03-21 at 3 12 53 pm](https://cloud.githubusercontent.com/assets/1204467/24173337/ea2d7b0a-0e48-11e7-9853-9917b6a41103.png)

I also updated it to set template.visibility to zero (all migrated records are zero but new ones are NULL).

**Comment by briri** [comment]
Fixed the above issue where the phase was not editable by updating the controller's new line to create a default phase so that its modifiable flag is true.

**Comment by briri** [comment]
Removed auto-phase generation line and instead updated the view so that the user sees the 'Add new phase' tab. It was checking whether the user was an org admin and then whether the org associated with the template or the user was designated as a funder. This prevents institutions from creating phases on their own templates. 

So I added another check to allow the tab to appear for users if their org matches the org associated with the template (meaning its their org's template)

![screen shot 2017-03-22 at 12 09 18 pm](https://cloud.githubusercontent.com/assets/1204467/24216274/d84e60c4-0ef8-11e7-8fd8-a29a1000ac30.png)

**Comment by briri** [comment]
These changes are out on the test server and have already been sent through as part of a PR. Just neglected to move this issue along in the project board

**Comment by sjDCC** [comment]
This all works fine so thanks @briri for the fixes. Only difference is that it doesn't automatically create/open a section once a phase has been created. We did that for usability to make people realize where to click next. Some people had been adding more phases, thinking they were sections in a template and not looking down (rather than across) if that makes sense.

Have included a screengrab to show what I mean. There are two places where people could click after creating a phase and the 'add new phase' button is more obvious / inviting so we perhaps want to draw them down to 'add section' We can do that through training guides for now though. It all works

![capture](https://cloud.githubusercontent.com/assets/6169940/24515592/fe4b2942-156f-11e7-9e3d-4e44f21d0108.PNG)

**Comment by stephaniesimms** [comment]
I receive a message that I'm "not authorized to perform this action" when I attempt to create a template. @briri or @sjDCC can you double check that this is working and/or check my permissions. I appear to have admin privileges so don't know why I can't create a template.

**Comment by sjDCC** [comment]
I am getting the same error now and could create a template before. Should this be moved back or start a new ticket about permissions?

**Comment by stephaniesimms** [comment]
@briri move this issue back to an active position on the board or create a new issue?

**Comment by briri** [comment]
I ended up modifying at what point the authorization check was happening when I built out the test for this. The test was sending in an org id but that's not sent in by the form and so is not available at the time the authorization check was happening.

Updated the test so that it mimics what is being passed back from the form. The removed the check that ensures the logged in user's org matches the templates org. The code always uses the user's org when creating a template even if one is submitted during the request.
